lib/dialects/mysql/query-generator.js

```diff
@@ -148,8 +148,12 @@ class MySQLQueryGenerator extends AbstractQueryGenerator {
     return `SHOW FULL COLUMNS FROM ${table};`;
   }
 
-  showTablesQuery() {
-    return "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = 'BASE TABLE';";
+  showTablesQuery(database) {
+    let query = 'SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA NOT IN (\'MYSQL\', \'INFORMATION_SCHEMA\', \'PERFORMANCE_SCHEMA\') AND TABLE_TYPE = \'BASE TABLE\'';
```

Comment:
What if I want to query the `MYSQL` schema ?
Perhaps this check would be a bit better
```
if (database) {
  query += ' AND TABLE_SCHEMA = `${database}`'
} else {
  query += ' AND TABLE_SCHEMA NOT IN ...'
}
```